### PR TITLE
test(promutil): add promutil simulator test

### PR DESCRIPTION
**Pull request by maxshuang**
add simulator test for multi-projects metric test

### CODE REVIEW

File: pkg/promutil/test/multiprojects_metric_scences.go [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+	go func() {
+		defer wg.Done()
+		log.L().Info("Start http listen on :8083")
+		http.ListenAndServe(":8083", nil)
+	}()
+
+	go func() {
+		defer wg.Done()
+		log.L().Info("Start scenarios simulator")
+		simulator(wg)
```
Comment (amyangfei) [inline pkg/promutil/test/multiprojects_metric_scences.go:27]: If a `WaitGroup` is explicitly passed into functions, it must be done by pointer, such as discussion: https://youtrack.jetbrains.com/issue/GO-7472/WaitGroup-inspection-when-passing-wg-not-as-pointer

File: pkg/promutil/test/multiprojects_metric_scences.go [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+			ConstLabels: prometheus.Labels{
+				"service": "svr1",
+			},
+		})
+		counter.Add(1)
+
+		for {
+			counter.Add(0.1)
+			time.Sleep(time.Second)
+		}
```
Comment (amyangfei) [inline pkg/promutil/test/multiprojects_metric_scences.go:96]: Better to release the goroutine.

File: pkg/promutil/test/go.mod [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	golang.org/x/time v0.0.0-20220224211638-0e9765cccd65 // indirect
+	google.golang.org/genproto v0.0.0-20220216160803-4663080d8bc8 // indirect
+	google.golang.org/grpc v1.44.0 // indirect
+	google.golang.org/protobuf v1.27.1 // indirect
+	gopkg.in/natefinch/lumberjack.v2 v2.0.0 // indirect
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
+	gorm.io/gorm v1.23.4 // indirect
+)
+
+replace github.com/hanfei1991/microcosm v0.0.0-20220519065215-9a7a74c24c30 => ../../../../microcosm
```
Comment (amyangfei) [inline pkg/promutil/test/go.mod:76]: Is this replace necessary

File: pkg/promutil/test/multiprojects_metric_scences.go [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+	"sync"
+	"time"
+
+	"github.com/hanfei1991/microcosm/pkg/promutil"
+	"github.com/hanfei1991/microcosm/pkg/tenant"
+	"github.com/pingcap/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func main() {
```
Comment (amyangfei) [inline pkg/promutil/test/multiprojects_metric_scences.go:14]: Do we need to run this test in CI?

File: pkg/promutil/test/multiprojects_metric_scences.go (hunk 2 above)
Comment (maxshuang) [reply pkg/promutil/test/multiprojects_metric_scences.go:96]: This main test is used for continous simulation test.
User can use `CTRL+C` to terminate the test.
Seems ok to me.

File: pkg/promutil/test/go.mod (hunk 3 above)
Comment (maxshuang) [reply pkg/promutil/test/go.mod:76]: I will remove it.
After merge to master, it's not necessary.

File: pkg/promutil/test/multiprojects_metric_scences.go (hunk 4 above)
Comment (maxshuang) [reply pkg/promutil/test/multiprojects_metric_scences.go:14]: Seems unnecessary.
I use this test to observe whether promethus and grafana can work normally.
I don't know how to check the promethus and grafana if we run this test in CI and what we expect from this ci.

**Review by amyangfei — APPROVED** [review]

**Pull request merged**